src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessible.java

```diff
@@ -182,7 +182,8 @@ public void propertyChange(PropertyChangeEvent e) {
 
                     // Do send check box state changes to native side
                     if (thisRole == AccessibleRole.CHECK_BOX) {
-                        if (newValue != null && !newValue.equals(oldValue)) {
+                        if ((newValue != null && !newValue.equals(oldValue)) ||
+                                oldValue != null && !oldValue.equals(newValue)) {
```

Comment:
At first glance, these conditions look like the same thing. equals() above will return false if oldValue is null. oldValue.equals(newValue) and newValue.equals(oldValue) are also the same thing. Try to rewrite this condition more clearly.

Reply:
When CheckBox state is `checked`, `newValue is Checked` and `oldValue is null` and the existing condition `if (newValue != null && !newValue.equals(oldValue))` works well. But when CheckBox is `unchecked`, `newValue is null` and `oldValue is checked`, the existing condition returns `false` when `newValue != null` is evaluated.

oldValue.equals(newValue) and newValue.equals(oldValue) looks same but when newValue is null, newValue.equals(oldValue) throws NPE.

So, I think these conditions are not exactly same. We need to evaluate the `oldValue!=null` when CheckBox is unchecked for the correct behaviour.

@aivanov-jdk suggested simpler way for comparing oldValue and newValue [here](https://github.com/openjdk/jdk/pull/23436#discussion_r1941484674). I will check and update.

Reply:
I think @savoptik meant that `!oldValue.equals(newValue)` was redundant.

Reply:
Yes,` oldValue.equals(newValue)` or `newValue.equals(oldValue)` always return same value. But when `newValue` was null, I couldn't use `newValue.equals(oldValue)` as it throws NPE. Anyway, the condition is simplified now.